pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -218,6 +246,24 @@ func NewDeploymentController(client kube.Client, clusterID cluster.ID, env *mode
 	dc.services.AddEventHandler(parentHandler)
 	dc.clients[gvr.Service] = NewUntypedWrapper(dc.services)
 
+	dc.configMaps = kclient.NewFiltered[*corev1.ConfigMap](client, filter)
+	dc.configMaps.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
+		// This could be a configmap referenced by a Gateway paramsRef
+		impacted := gatewaysByParamsRef.Lookup(config.NamespacedName(o))
+		for _, gw := range impacted {
+			dc.queue.AddObject(gw)
+		}
+		// Or it could also be a global GatewayClass config
+		classDefaults, classDefaultsF := o.GetLabels()[gatewayClassDefaults]
```

Comment:
How does this deal with duplicates (multiple configmaps with the same label)? Looks like this could perhaps cause unnecessary reconciliation

Reply:
Are you suggesting something like if we have duplicates, 1 will be "dead" so we should ignore it and not reconcile? 

If so - I think that's technically a gap but not something worth caring about. A reconcile is fairly cheap, and the likelihood someone is constantly going to be changing an invalid config map seems quite low. In general (without krt) i think it's best to focus on simplicity and correctness for this type of logic, except for areas where that has proven to be a performance issue

Reply:
If you just mean in general - that is in the reconcile function